homeassistant/components/songpal/media_player.py

```diff
@@ -98,17 +108,31 @@ async def async_setup_entry(
         {vol.Required(PARAM_NAME): cv.string, vol.Required(PARAM_VALUE): cv.string},
         "async_set_sound_setting",
     )
+    platform.async_register_entity_service(
```

Comment:
Please use our standard media player api for groups:
https://developers.home-assistant.io/docs/core/entity/media-player#grouping-player-entities-together

Reply:
Thanks! `join` looks what I need indeed. However, `unjoin` looks like it should be used slave-initiated; which isn't supported by the SongPal API: it's only possible to break up a group from the perspective of the master. (So with a group of 3, you cannot remove 1, but only break up the whole group) Any guidance for that?

Reply:
Maybe only allow to unjoin in the master entity and raise an error in the other entities in the service handler?